includes/derivatives.inc

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/**
+ * @file
+ * Holds derivative functionality for compound solution pack.
+ */
+
+/**
+ * Derivative creation function for compound parent object TN.
+ *
+ * Wraps the TN update/creation function and
+ * adds expected logging.
+ *
+ * @param AbstractObject $object
+ *   The compound object.
+ * @param bool $force
+ *   Boolean for forcing this derivative creation.
+ *
+ * @return array
+ *   Success message.
+ */
+function islandora_compound_object_create_thumbnail(AbstractObject $object, $force = TRUE) {
+  if ($force || !isset($object['TN'])) {
+    $parts = islandora_compound_object_get_parts($object->id);
+    if (!empty($parts)) {
+      $child = islandora_object_load($parts[0]);
+      islandora_compound_object_update_parent_thumbnail($object, $child);
+    }
+    if (isset($child) && $object['TN']->size == $child['TN']->size) {
```

Comment:
I'm not sure what's being described with:

> And wrap islandora_compound_object_update_parent_thumbnail's logic in try catch?

But yeah, the [example code there](https://github.com/Islandora/islandora_solution_pack_compound/pull/126#discussion_r123256980) is much what I might expect.

In `islandora_compound_object_update_parent_thumbnail()`, I was [just thinking with](https://github.com/Islandora/islandora_solution_pack_compound/pull/126#discussion_r123238755):

> that content-checking-bit beforehand could almost be shoved into the islandora_compound_object_update_parent_thumbnail()... would probably want to add something to the comment to indicate that it won't do anything if the datastream content is the same.

so, before setting the content, to check if we'd actually be introducing a change (other than creating a new version) if we were to proceed and set the content from the child. In this change, we'd want the fail-fast behaviour, checking the `size` property (and possibly others? `mimetype`? `checksumType` and `checksum` (if they're set)?), before finally comparing the `content`, so if any are _different_, the more expensive `content` comparison might be avoided.